upi/openstack/network.yaml

```diff
@@ -34,6 +76,86 @@
       ipv6_ra_mode: "{{ os_subnet6_router_advertisements_mode }}"
     when: os_subnet6_range is defined
 
+  - name: 'List the cluster address-scope'
+    command:
+      cmd: "openstack address scope list --name node_network_address_scope"
+    when: os_networking_type == "CiscoACI"
+    register: cluster_address_scope
+
+  - name: 'Create the cluster address-scope'
+    command:
+      cmd: "neutron address-scope-create node_network_address_scope 4 --apic:distinguished_names type=dict VRF={{ aci_cni['network_interfaces']['node']['vrf'] }}"
+    when:
+      - os_networking_type == "CiscoACI"
+      - cluster_address_scope.stdout == ""
+
+  - name: 'list subnet pool'
+    command:
+      cmd: "openstack subnet pool list --name node_network_subnet_pool -f value -c Prefixes"
+    when: os_networking_type == "CiscoACI"
+    register: primary_subnet_pool
+
+  - name: 'Check if subnetpool already exists with different subnet range.'
+    assert:
+      that:
+        - os_subnet_range in primary_subnet_pool.stdout
+      fail_msg: "node_network_subnet_pool already exists with different subnet range."
+    when:
+      - os_networking_type == "CiscoACI"
+      - primary_subnet_pool.stdout != ""
+
+  - name: 'Create the subnetpool'
+    command:
+      cmd: "neutron subnetpool-create --pool-prefix {{ os_subnet_range }} --address-scope node_network_address_scope node_network_subnet_pool"
+    when:
+      - os_networking_type == "CiscoACI"
+      - primary_subnet_pool.stdout == ""
+
+  - name: 'list primary subnet'
+    command:
+      cmd: "openstack subnet list --name {{ os_subnet }} --network {{ os_network }} --tag {{ cluster_id_tag }}"
+    when: os_networking_type == "CiscoACI"
+    register: primary_cluster_subnet
+
+  - name: 'get cidr of primary subnet'
+    command:
+      cmd: "openstack subnet list --name {{ os_subnet }} --network {{ os_network }} --tag {{ cluster_id_tag }} -f value -c Subnet"
+    when:
+      - os_networking_type == "CiscoACI"
+      - primary_cluster_subnet.stdout != ""
+    register: primary_subnet_cidr
+
+  - name: 'Check if primary cluster subnet already exists with different subnet range.'
+    assert:
+      that:
+        - os_subnet_range in primary_subnet_cidr.stdout_lines|list
+      fail_msg: "{{ os_subnet }} already exists with different subnet range."
+    when:
+      - os_networking_type == "CiscoACI"
+      - primary_cluster_subnet.stdout != ""
+
+  - name: 'Create the primary cluster subnet'
+    command:
+      cmd: "openstack subnet create --network {{ os_network }}  --subnet-pool node_network_subnet_pool --subnet-range {{ os_subnet_range }} --allocation-pool start={{ os_subnet_range | next_nth_usable(10) }},end={{ os_subnet_range | ipaddr('last_usable') }} --dhcp {{ os_subnet }}"
+    when:
+      - os_networking_type == "CiscoACI"
+      - primary_cluster_subnet.stdout == ""
+
+  - name: 'Create the cluster IPv6 subnet'
```

Comment:
This entry to create IPv6 subnet should be removed as we already defined above.

Reply:
Thanks @maheshkurund I have rectified it.
For security group issue - we have to remove it from kmr2-release-4.16 as well as i can see same changes on 4.16 1. https://github.com/noironetworks/openshift_installer/pull/203/commits/ad3dcb5411557b1825ba27573687c91549a387eb#diff-810ef9226dc20e51f9b2814e9885d243be1590bff38badb8d2b341263915d30cL194-L195
2. https://github.com/noironetworks/openshift_installer/pull/203/commits/ad3dcb5411557b1825ba27573687c91549a387eb#diff-810ef9226dc20e51f9b2814e9885d243be1590bff38badb8d2b341263915d30cL226-L227